src/hooks/useGetProviderEServiceTemplateActions.ts

```diff
@@ -0,0 +1,302 @@
+import type {
+  EServiceDescriptorState,
+  EServiceMode,
+  EServiceTemplateVersionState,
+} from '@/api/api.generatedTypes'
+import { useNavigate } from '@/router'
+import { useTranslation } from 'react-i18next'
+import type { ActionItemButton } from '@/types/common.types'
+import { AuthHooks } from '@/api/auth'
+import FiberNewIcon from '@mui/icons-material/FiberNew'
+import PlayCircleOutlineIcon from '@mui/icons-material/PlayCircleOutline'
+import PauseCircleOutlineIcon from '@mui/icons-material/PauseCircleOutline'
+import ContentCopyIcon from '@mui/icons-material/ContentCopy'
+import DeleteOutlineIcon from '@mui/icons-material/DeleteOutline'
+import CheckCircleOutlineIcon from '@mui/icons-material/CheckCircleOutline'
+import PendingActionsIcon from '@mui/icons-material/PendingActions'
+import PublishIcon from '@mui/icons-material/Publish'
+import { useDialog } from '@/stores'
+import { match } from 'ts-pattern'
+import { TemplateMutations } from '@/api/template'
+
+export function useGetProviderEServiceTemplateActions(
+  eServiceTemplateId: string,
+  eServiceTemplateVersionId: string,
+  mode: EServiceMode | undefined, //TODO
+  activeVersionState?: EServiceTemplateVersionState | undefined,
+  draftVersionState?: EServiceTemplateVersionState | undefined
+): { actions: Array<ActionItemButton> } {
+  const { t } = useTranslation('common', { keyPrefix: 'actions' })
+
+  const { isAdmin, isOperatorAPI, jwt } = AuthHooks.useJwt()
+  const navigate = useNavigate()
+  const { openDialog, closeDialog } = useDialog()
+
+  const { mutate: publishDraft } = TemplateMutations.usePublishVersionDraft()
+  const { mutate: deleteVersionDraft } = TemplateMutations.useDeleteVersionDraft()
+  const { mutate: suspend } = TemplateMutations.useSuspendVersion()
+  const { mutate: reactivate } = TemplateMutations.useReactivateVersion()
+  const { mutate: createNewDraft } = TemplateMutations.useCreateDraft()
+
+  const state = activeVersionState ?? 'DRAFT'
+  const hasVersionDraft = state === 'DRAFT' || draftVersionState === 'DRAFT'
+
+  // Only admin and operatorAPI can see actions
+  if (!isAdmin && !isOperatorAPI) return { actions: [] }
+
+  const handlePublishDraft = () => {
+    if (state === 'DRAFT')
```

Comment:
I would remove this check since this function should trigger only if the template is DRAFT.
If it somehow triggers when the template is not DRAFT, it is better to have the FE to call the `publishDraft`, that will produce an error that will tell us that something is wrong with our business logic

Reply:
I know the early returns is something that are pretty much everywhere in the codebase, but that is something I learnt to regret